### PR TITLE
allow passing file path as reference to devicelab task

### PR DESCRIPTION
**Pull request by yjbanov**
/cc @Hixie

### CODE REVIEW

File: dev/devicelab/README.md [hunk 1]
```diff
@@ -26,10 +28,16 @@ To run a test, use option `-t` (`--task`):
...
 ```
 
-You can see the test names in the `manifest.yaml` file in this directory or by looking in `bin/tasks/`. Do not include the `.dart` file extension. For example, `dart bin/run.dart -t complex_layout__start_up`.
+Where `NAME_OR_PATH_OF_TEST` can be either of:
+
+- the _name_ of a task, which you can find in the `manifest.yaml` file in this
+  directory. Example: `complex_layout__start_up`.
+- the path to a Dart _file_ corresponding to a task, which resides in `bin/tasks`.
+  Tip: most shells support path auto-completion using the Tab key. Example:
+  `bin/tasks/complex_layout__start_up.dart`.
```
Comment (Hixie) [inline dev/devicelab/README.md]: how do you disambiguate? these namespaces overlap...

File: dev/devicelab/bin/run.dart [hunk 2]
```diff
@@ -73,9 +75,27 @@ final ArgParser _argParser = new ArgParser()
...
+      for (String nameOrPath in value) {
+        List<String> fragments = path.split(nameOrPath);
+        bool isDartFile = fragments.last.endsWith('.dart');
+
+        if (fragments.length == 1 && !isDartFile) {
+          // Not a path
+          _taskNames.add(nameOrPath);
+        } else if (!isDartFile || fragments.length != 3 || !_listsEqual(<String>['bin', 'tasks'], fragments.take(2).toList())) {
+          // Unsupported executable location
+          throw new FormatException('Invalid value for option -t (--task): $nameOrPath');
```
Comment (Hixie) [inline dev/devicelab/bin/run.dart]: it's unusual to accept a path but only allow certain paths... what if i want to test a version in my /tmp/ directory?

Comment (Hixie) [inline dev/devicelab/bin/run.dart]: in any case if you keep this restriction the message should say what the restriction is

**Comment by Hixie** [comment]
LGTM

File: dev/devicelab/README.md (hunk 1 above)
Comment (yjbanov) [inline dev/devicelab/README.md]: Task Dart files must go in `bin/tasks` and have a `.dart` extension. Names have no slashes or extensions.

File: dev/devicelab/bin/run.dart (hunk 2 above)
Comment (yjbanov) [inline dev/devicelab/bin/run.dart]: `FormatException`s are accompanied by usage printout, which explains the restriction.

Alternatively, we could have a separate option for file paths (`-f` maybe?), lift the `bin/tasks` restriction and simplify the logic (and docs).

LMK which you prefer. It's an easy change.

Comment (Hixie) [inline dev/devicelab/bin/run.dart]: Either's fine by me. Two options means you now have to decide which one takes precedence, or catch both being set, etc. So it's not necessarily a win.

**Pull request merged**